pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/util/ClassUtils.java

```diff
@@ -97,33 +98,45 @@
                 } else {
                     log.debug("Scanning filesystem {} for classes", packageUrl);
                     log.debug(packageUrl.getProtocol());
-                    try (InputStream is = packageUrl.openStream()) {
-                        Objects.requireNonNull(is, String.format("Package url %s stream is null!", packageUrl));
-                        try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
-                            classes.addAll(reader.lines()
-                                    .filter(line -> line != null && line.endsWith(".class"))
-                                    .map(line -> {
-                                        String className = line.replace(".class", "").replace('/', '.');
-                                        try {
-                                            return getClass(packagePrefix + className);
-                                        } catch (Exception e) {
-                                            log.error("Failed to find {}", line, e);
-                                        }
-
-                                        return null;
-                                    })
-                                    .collect(Collectors.toList()));
-                        }
-                    } catch (IOException e) {
-                        throw new PippoRuntimeException("Failed to get classes for package '{}'", e, packageName);
-                    }
+//                    try {
+                        classes.addAll(
+                            getAllFiles(packageUrl, ".class").stream()
+                            .map(filePath -> {
+                                filePath = filePath.replace(File.separatorChar, '.').replaceFirst(".class$", "");
+                                return packagePrefix + filePath.split(packagePrefix, 2)[1];
+                            })
+                            .map(className -> {
+                                try {
+                                    return getClass(className);
+                                } catch (Exception e) {
+                                    log.error("Failed to find {}", className, e);
+                                }
+                                return null;
+                            })
+                            .collect(Collectors.toList())
+                        );
+//                    } catch (Exception e) {
+//                        throw new PippoRuntimeException(e, "Failed to get classes for package '{}'", packageName);
+//                    }
                 }
             }
         }
 
+        // TODO: why unmodifiableCollection ?!
         return Collections.unmodifiableCollection(classes);
     }
 
+    private static List<String> getAllFiles(URL packageUrl, String endsWith) {
```

Comment:
If this private method is not used for other purpose (in other places), I propose you to rename it to "getClassFiles" and to remove the `endsWith` parameter.